server/src/future.rs

```diff
@@ -94,12 +85,6 @@ where
 			}
 		}
 	}
-
-	fn poll_stop_monitor_heartbeat(&mut self, cx: &mut Context) {
-		// We don't care about the ticks of the heartbeat, it's here only
-		// to periodically wake the `Waker` on `cx`.
-		let _ = self.stop_monitor_heartbeat.poll_tick(cx);
```

Comment:
IIUC we relied on this to make sure that we wake up:
- when there's no scheduled tasks that would yield to the executor (call waker.wake())
- for at most one second 

To remove `tokio::time::Interval` we are polling the `FutureDriver` alongside other operations that we do in the server:
- try_accept_conn: Await incoming connection
- wait_for_permit: Await backpressure for some internal buffer space
- try_recv: Can wait for `soketto::Pong`, `soketto::Data`,  and `soketto::Closed`

Could we still get in a situation where we have no other competing tasks to drive the `FutureDriver`?
(ie, we don't have `soketto::Pong` to wake us up -- or is configured at 100seconds -- and we don't receive `soketto::Data`)

Reply:
After convincing myself that this DriverSelect thing (ie `select_with`) will poll the futures in itself as needed, maybe I'm missing something but why does it matter if this thing doesn't wake up for a while if none of the futures in it need to progress?

Reply:
so, that's good question/observation but I found that quite unlikely because both batch and calls are executed as one future/task.

Then in each loop iteration both `try_recv` and `wait_for_permit` are awaited on which checks "future vec" but not sure if there is possible to spawn huge futures and then "never send some data again/backpressure kicks in" then those would never woken up again but I don't see it as an issue because it's to the client enforce that.

If it's idle fine but we could have a few ready tasks in the vec.